portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,13 +29,10 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> comments;
+  private ArrayList<Comment> comments;
 
   public DataServlet() {
-    comments = new ArrayList<>(3);
-    comments.add("Hello!");
-    comments.add("Nice to meet you!");
-    comments.add("Have a nice day!");
+    comments = new ArrayList<>(5);
```

Comment:
Don't set the capacity of the array list unless you're immediately adding contents to it. Now comments is being grown dynamically in doPost